probe/endpoint/procspy/proc_linux.go

```diff
@@ -23,6 +24,23 @@ var (
 	netNamespacePathSuffix = ""
 )
 
+type pidWalker struct {
+	walker      process.Walker
+	ticker      <-chan time.Time
```

Comment:
Also, it would be good to have a comment somewhere indicating this is for rate limiting of individual namespaces.